src/backend/data/feed.js

```diff
@@ -37,7 +44,34 @@ class Feed {
    * Returns a Promise.
    */
   save() {
-    addFeed(this);
+    return addFeed(this);
+  }
+
+  /**
+   * Removes current Feed + associated Posts from the databases.
+   * Returns a Promise
+   */
+  async delete() {
```

Comment:
This function has a lot of Promises, which is not great for performance.  We should see if we can reduce the number in a follow-up PR.